code/game/gamemodes/cult/runes.dm

```diff
@@ -433,590 +340,610 @@ var/list/teleport_runes = list()
-					to_chat(G, "<span class='cultitalic'>You feel the link between you and your body weakening... you must hurry!</span>")
-		if(user.stat == DEAD)
-			user.color = initial(user.color)
-			rune_in_use = 0
-			affecting = null
-			var/mob/dead/observer/G = user.get_ghost()
-			if(G)
-				to_chat(G, "<span class='cultitalic'><b>You suddenly feel your physical form pass on. [src]'s exertion has killed you!</b></span>")
-			return
-		sleep(10)
-	rune_in_use = 0
-
-
 //Rite of the Corporeal Shield: When invoked, becomes solid and cannot be passed. Invoke again to undo.
 /obj/effect/rune/wall
-	cultist_name = "Rite of the Corporeal Shield"
-	cultist_desc = "when invoked, makes an invisible wall to block passage. Can be invoked again to reverse this."
+	cultist_name = "Barrier"
+	cultist_desc = "when invoked, makes a temporary invisible wall to block passage. Will chain-invoke nearby unactive barrier runes. Can be invoked again to reverse this."
 	invocation = "Khari'd! Eske'te tannin!"
-	icon_state = "1"
-	invoke_damage = 2
+	icon_state = "barrier"
+	var/datum/timedevent/density_timer
+	var/recharging = FALSE
 
-/obj/effect/rune/wall/examine(mob/user)
+/obj/effect/rune/wall/Initialize(mapload)
 	. = ..()
-	if(density)
-		. += "<span class='cultitalic'>There is a barely perceptible shimmering of the air above [src].</span>"
+	GLOB.wall_runes += src
+
+/obj/effect/forcefield/CanAtmosPass(turf/T)
+	return !density
+
+/obj/effect/rune/wall/Destroy()
+	GLOB.wall_runes -= src
+	return ..()
 
 /obj/effect/rune/wall/invoke(var/list/invokers)
+	if(recharging)
+		return
 	var/mob/living/user = invokers[1]
 	..()
 	density = !density
-	user.visible_message("<span class='warning'>[user] places [user.p_their()] hands on [src], and [density ? "the air above it begins to shimmer" : "the shimmer above it fades"].</span>", \
-						 "<span class='cultitalic'>You channel your life energy into [src], [density ? "preventing" : "allowing"] passage above it.</span>")
-	if(iscarbon(user))
+	update_state()
+	if(density)
+		spread_density()
+	var/carbon_user = iscarbon(user)
+	user.visible_message("<span class='warning'>[user] [carbon_user ? "places [user.p_their()] hands on":"stares intently at"] [src], and [density ? "the air above it begins to shimmer" : "the shimmer above it fades"].</span>", \
+						 "<span class='cultitalic'>You channel [carbon_user ? "your life ":""]energy into [src], [density ? "temporarily preventing" : "allowing"] passage above it.</span>")
+	if(carbon_user)
 		var/mob/living/carbon/C = user
-		C.apply_damage(2, BRUTE, pick("l_arm", "r_arm"))
-
+		C.apply_damage(2, BRUTE, pick(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM))
+
+/obj/effect/rune/wall/proc/spread_density()
+	for(var/R in GLOB.wall_runes)
+		var/obj/effect/rune/wall/W = R
+		if(W.z == z && get_dist(src, W) <= 2 && !W.density && !W.recharging)
+			W.density = TRUE
+			W.update_state()
+			W.spread_density()
+	density_timer = addtimer(CALLBACK(src, .proc/lose_density), 3000, TIMER_STOPPABLE)
```

Comment:
This one doesn't quite fit in here. spread_density suggests it is spreading the density to other runes. Not that it also sets the kill timer

Reply:
This is here since the proc is called recursively. Will rename for clarity.